starter/main.cc

```diff
@@ -26,6 +27,16 @@ int main(int /* argc */, char** argv) {
     exit(1);
   }
 
+  bool keepCapNetBindService = false;
+  for (int i = 0; i < argc; ++i) {
+    if (std::string(argv[i]) == "--keep-cap-net-bind-service") {
+      // keep CAP_NET_BIND_SERVICE if it's present in the effective capabilities
+      keepCapNetBindService = (caps & (1UL << CAP_NET_BIND_SERVICE)) != 0;
+      // remove (reset) argument (not passing to Envoy process)
+      argv[i] = new char[0];
+    }
```

Comment:
Good idea

I implemented it the way that the separator `--` is mandatory - even in an edge case where no arguments for envoy follow. Yes this is a breaking change from a usage perspective. 

But it would also be a breaking change the other way around. Because without the separator (how it is currently configured in cilium/cilium) - after the change, all arguments would belong to the starter itself (whereas previously they would be passed to Envoy).

Reply:
IMO `--` should be optional. If not present maybe just replace `argv[0]` with the path, or copy everything over etc. A bit more work, but much nicer in the case no starter options are used, I think.

Reply:
agree. that's why i dropped the conditional abort if `--` isn't present. https://github.com/cilium/proxy/pull/650#discussion_r1553728139

Reply:
as discussed, i added backwards compatibility by handling all arguments as Envoys if the separator `--` isn't passed to the starter.